File: backend/app/schemas/common_code.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class CodeGroupItem(BaseModel):
    id: int
    code: str
    name: str
    description: str | None
    is_active: bool
    sort_order: int
    created_at: datetime
    updated_at: datetime


class CodeItem(BaseModel):
    id: int
    group_id: int
    code: str
    name: str
    description: str | None
    is_active: bool
    sort_order: int
    extra_value1: str | None
    extra_value2: str | None
    created_at: datetime
    updated_at: datetime


class CodeGroupListResponse(BaseModel):
    groups: list[CodeGroupItem]


class CodeGroupDetailResponse(BaseModel):
    group: CodeGroupItem


class CodeListResponse(BaseModel):
    codes: list[CodeItem]


class CodeDetailResponse(BaseModel):
    code: CodeItem


class ActiveCodeOption(BaseModel):
    code: str
    name: str


class ActiveCodeListResponse(BaseModel):
    group_code: str
    options: list[ActiveCodeOption]


class CodeGroupCreateRequest(BaseModel):
    code: str = Field(min_length=1, max_length=30)
    name: str = Field(min_length=1, max_length=100)
    description: str | None = None
    is_active: bool = True
    sort_order: int = 0


class CodeGroupUpdateRequest(BaseModel):
    code: str | None = Field(default=None, min_length=1, max_length=30)
    name: str | None = Field(default=None, min_length=1, max_length=100)
    description: str | None = None
    is_active: bool | None = None
    sort_order: int | None = None


class CodeCreateRequest(BaseModel):
    code: str = Field(min_length=1, max_length=30)
    name: str = Field(min_length=1, max_length=100)
    description: str | None = None
    is_active: bool = True
    sort_order: int = 0
    extra_value1: str | None = Field(default=None, max_length=200)
    extra_value2: str | None = Field(default=None, max_length=200)


class CodeUpdateRequest(BaseModel):
    code: str | None = Field(default=None, min_length=1, max_length=30)
    name: str | None = Field(default=None, min_length=1, max_length=100)
    description: str | None = None
    is_active: bool | None = None
    sort_order: int | None = None
    extra_value1: str | None = Field(default=None, max_length=200)
    extra_value2: str | None = Field(default=None, max_length=200)

File: frontend/src/types/common-code.ts
export type CodeGroupItem = {
  id: number;
  code: string;
  name: string;
  description: string | null;
  is_active: boolean;
  sort_order: number;
  created_at: string;
  updated_at: string;
};

export type CodeItem = {
  id: number;
  group_id: number;
  code: string;
  name: string;
  description: string | null;
  is_active: boolean;
  sort_order: number;
  extra_value1: string | null;
  extra_value2: string | null;
  created_at: string;
  updated_at: string;
};

export type CodeGroupListResponse = { groups: CodeGroupItem[] };
export type CodeGroupDetailResponse = { group: CodeGroupItem };
export type CodeListResponse = { codes: CodeItem[] };
export type CodeDetailResponse = { code: CodeItem };
export type ActiveCodeOption = { code: string; name: string };
export type ActiveCodeListResponse = { group_code: string; options: ActiveCodeOption[] };
